worker/codalabworker/bundle_service_client.py

```diff
@@ -21,15 +20,18 @@ def wrapper(*args, **kwargs):
             except BundleAuthException:
                 raise
             except RestClientException as e:
-                raise BundleServiceException(message + ': ' + e.message, e.client_error)
-            except urllib2.HTTPError as e:
-                client_error = json.load(e)
+                raise BundleServiceException(message + ': ' + str(e), e.client_error)
+            except urllib.error.HTTPError as e:
+                try:
+                    client_error = json.loads(e.read().decode())
+                except json.JSONDecodeError:
+                    client_error = {'error': e.read().decode()}
```

Comment:
why are we decoding these strings?

Reply:
@percyliang urllib2.HTTPError returns a file-like object which is opened in **binary** mode. Thus, `json.load` doesn't work on it, so we have to read and convert it to a string first.